src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const separatorToString = String(separator);
+    let newJoin = '';
+    const lastSymbol = this[this.length - 1];
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        newJoin += this[i] + separatorToString;
+      } else {
+        newJoin += separatorToString;
+      }
+    }
+
+    if (lastSymbol !== null && lastSymbol !== undefined) {
+      newJoin += lastSymbol;
```

Comment:
I think it should work without this one "lastSymbol !== null &&" it works just with lastSymbol !== undefined

Reply:
you're right, fixed :)